stablehlo/dialect/TypeInference.h

```diff
@@ -268,13 +329,22 @@ LogicalResult inferTupleOp(MLIRContext* context, Optional<Location> location,
                            ValueRange val,
                            SmallVectorImpl<Type>& inferredReturnTypes);
 
+LogicalResult inferUniformDequantizeOp(
+    Optional<Location> location, Value operand,
+    SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes);
+
 LogicalResult inferWhileOp(Optional<Location> location, ValueRange operand,
                            SmallVectorImpl<Type>& inferredReturnTypes);
 
```

Comment:
There's one shape function which hasn't yet moved to TypeInference.cpp and is not in the TODO list in the PR description: `ImagOp::inferReturnTypes`. Can you handle it in this PR as well?